File: benchmark/comparison/results/haiku_py/03.py
class Color:
    def __init__(self, name, hex_code):
        self.name = name
        self.hex_code = hex_code

    def to_hex(self):
        return self.hex_code

    def is_primary(self):
        return self.name in ["Red", "Green", "Blue"]

# Create colors
red = Color("Red", "#FF0000")
green = Color("Green", "#00FF00")
blue = Color("Blue", "#0000FF")
yellow = Color("Yellow", "#FFFF00")

colors = [red, green, blue, yellow]

# Print hex for all 4
for color in colors:
    print(color.to_hex())

# Print is_primary for all 4
for color in colors:
    print("true" if color.is_primary() else "false")

File: benchmark/comparison/results/adv_ts/03.ts
type Color = "Red" | "Green" | "Blue" | "Yellow";

function to_hex(c: Color): string {
  switch (c) {
    case "Red": return "#FF0000";
    case "Green": return "#00FF00";
    case "Blue": return "#0000FF";
    case "Yellow": return "#FFFF00";
  }
}

function is_primary(c: Color): boolean {
  return c === "Red" || c === "Green" || c === "Blue";
}

const colors: Color[] = ["Red", "Green", "Blue", "Yellow"];
for (const c of colors) {
  console.log(to_hex(c));
}
for (const c of colors) {
  console.log(String(is_primary(c)));
}
